remote-view/shared/src/main/scala/parsley/debug/RemoteView.scala

```diff
@@ -0,0 +1,131 @@
+    val RESPONSE_TIMEOUT: FiniteDuration = 10.second
+
+    // Create localhost endpoint using port
+    val sb: StringBuilder = new StringBuilder("http://")
+    sb.append(address)
+    sb.append(':')
+    sb.append(port.toString())
+    sb.append("/remote")
+    val endPoint: String = sb.result()
+
+    // Format payload as JSON
+    sb.clear()
+    sb.append("{\"input\": \"")
+    sb.append(input)
+    sb.append("\", \"tree\": ")
+    sb.append(DebugTreeSerialiser.toJSON(tree))
+    sb.append("}")
+    val payload: String = sb.result()
+
+    println("Sending Debug Tree to Server")
+
+    // Send POST
+    try {
+      val backend = HttpURLConnectionBackend(
+        options = SttpBackendOptions.connectionTimeout(CONNECTION_TIMEOUT)
+      )
+      val r: Response[Either[String, String]] = basicRequest
+        .readTimeout(RESPONSE_TIMEOUT)
+        .header("User-Agent", "remoteView")
+        .contentType("application/json")
+        .body(payload)
+        .post(uri"$endPoint")
+        .send(backend)
+        
+        if (r.body.isLeft) {
+          println(s"${TEXT_TO_RED}Request Failed with message : ${r.body.getOrElse("<no-response>")}${TEXT_TO_NORMAL}")
+        } else {
+          println(s"Response from server : ${r.body.getOrElse("<no-response>")}")
+        }
+    } catch {
+      case _: Throwable => println(s"${TEXT_TO_RED}Remote View request failed! Please validate address ($address) and port number ($port).${TEXT_TO_NORMAL}")
+    }
+
+  }
+}
+
+object RemoteView extends DebugView.Reusable with RemoteView {
+    // Default port uses HTTP port and local host
+    override implicit val port = 80
+    override implicit val address = "127.0.0.1"
+    
+    /** Do some basic validations for a given IP address. */
+    private def checkIp(address: String): Boolean = {
+      val minimalAddress: String = "0.0.0.0"
+      val maximalAddress: String = "255.255.255.255"
+
+      if (address.length() < minimalAddress.length() || address.length() > maximalAddress.length()) {
+        return false
+      }
+      
+      if (address.count(_ == '.') != 3) {
+        return false
+      }
+      
+      // Check that every number is a number
+      val numberStrings: Array[String] = address.split('.')
+      numberStrings.forall((number: String) => number.length() > 0 && Try(number.toInt).toOption.nonEmpty)
+    }
+
+    /** Create a new instance of [[RemoteView]] with a given custom port. */
+    def apply(user_port: Integer = port, user_address: String = address): RemoteView = new RemoteView {
+      private val MAX_USER_PORT: Integer = 65535
+      assert(user_port <= MAX_USER_PORT, s"Remote View port invalid : $user_port > $MAX_USER_PORT")
+      assert(checkIp(user_address), s"Remote View address invalid : $user_address")
+
+      override implicit val port = user_port
+      override implicit val address = user_address
+    }
+}
+
+/** Helper object for connecting to the DILL backend. */
+object DillRemoteView extends RemoteView {
```

Comment:
```suggestion
object DillRemoteView extends DebugView.Reusable with RemoteView {
```
This helps the types in the documentation be clearer about what's going on

Reply:
👍

Reply:
👍